packages/app/src/containers/ComponentsWindow.tsx

```diff
@@ -0,0 +1,53 @@
+import { memo } from 'react';
+import ColorSwatch from '../components/swatches/ColorSwatch';
+import Sketch from '@sketch-hq/sketch-file-format-ts';
+import { GridView } from 'noya-designsystem';
+import { useApplicationState } from '../contexts/ApplicationStateContext';
+import { getSharedSwatches } from 'noya-state/src/selectors';
+import { rgbaToHex } from 'noya-colorpicker';
+
+export default memo(function ComponentsWindow() {
+  const [state, dispatch] = useApplicationState();
+
+  const items = getSharedSwatches(state);
+
+  const createSubtitle = (item: Sketch.Swatch) => {
```

Comment:
Since `item` has type `Sketch.Swatch`, it can never have a `_class` other than `'swatch'`, so I don't think we need to check for it.

In terms of coding style, I would do this within the `items.map`

E.g.

```ts
items.map((item, index) => {
  const color = {
    a: item.value.alpha,
    r: Math.round(item.value.red * 255),
    g: Math.round(item.value.green * 255),
    b: Math.round(item.value.blue * 255),
  };

  return (
    <GridView.Item
      key={index}
      title={item.name}
      subtitle={`${rgbaToHex(color)} - ${Math.round(color.a * 100)}%`}
      selected={state.selectedSwatchIds.includes(item.do_objectID)}
      onClick={(event) => {
        event.stopPropagation();
        event.preventDefault();

        dispatch(
          'selectSwatch',
          item.do_objectID,
          event.shiftKey ? 'intersection' : 'replace',
        );
      }}
    >
      <ColorSwatch value={item.value} />
    </GridView.Item>
  )
})
```